diff -ruN ../logstash.old/build.gradle ./build.gradle
--- ../logstash.old/build.gradle	2021-02-04 14:33:01.603528406 -0500
+++ ./build.gradle	2021-02-04 14:34:31.188294389 -0500
@@ -531,6 +531,25 @@
     }
 }
 
+def selectArch() {
+    if (project.ext.has("jdk_arch")) {
+        return project.ext.jdk_arch
+    }
+    String cpu_arch = System.properties["os.arch"]
+    switch (cpu_arch) {
+        case "amd64":
+        case "x86_64":
+            return "x86_64"
+        case "aarch64":
+        case "arm64":
+            return "arm64"
+        case "ppc64le":
+            return "ppc64le"
+        default:
+            throw new IllegalArgumentException("Can't handle os.arch of type $cpu_arch")
+    }
+}
+
 class JDKDetails {
     final String revision
     final String build
@@ -588,6 +607,9 @@
             case "arm64":
                 return "aarch64"
                 break
+            case "ppc64le":
+                return "ppc64le"
+                break
             default:
                 throw RuntimeException("Can't handle CPU architechture: ${jdkArch}")
         }
@@ -601,7 +623,7 @@
     String osName = selectOsType()
 
     def versionYml = new Yaml().load(new File("$projectDir/versions.yml").text)
-    String jdkArch = project.ext.jdk_arch
+    String jdkArch = selectArch()
     def jdkDetails = new JDKDetails(versionYml, osName, jdkArch)
 
     description "Download JDK ${jdkDetails.major}, OS: ${osName}"
diff -ruN ../logstash.old/rakelib/artifacts.rake ./rakelib/artifacts.rake
--- ../logstash.old/rakelib/artifacts.rake	2021-02-04 14:33:01.593527873 -0500
+++ ./rakelib/artifacts.rake	2021-02-04 14:35:26.501202559 -0500
@@ -206,8 +206,12 @@
     puts("[artifact:rpm] building rpm OSS package x86_64")
     package_with_jdk("centos", "x86_64", :oss)
 
-    puts("[artifact:rpm] building rpm OSS package arm64")
-    package_with_jdk("centos", "arm64", :oss)
+    #comment out arm64 packaging to speed things up
+    #puts("[artifact:rpm] building rpm OSS package arm64")
+    #package_with_jdk("centos", "arm64", :oss)
+
+    puts("[artifact:rpm] building rpm OSS package ppc64le")
+    package_with_jdk("centos", "ppc64le", :oss)
 
     #without JDKs
     system("./gradlew bootstrap") #force the build of Logstash jars
@@ -688,6 +692,8 @@
         else
           return "aarch64"
       end
+    elsif jdk_arch == 'ppc64le'
+      return "ppc64le"
     else
       raise "CPU architecture not recognized: #{jdk_arch}"
     end
